src/Stripe.Tests.XUnit/sources/listing_sources_on_customer.cs

```diff
@@ -0,0 +1,101 @@
+using FluentAssertions;
+using Xunit;
+
+namespace Stripe.Tests.Xunit
+{
+    public class listing_sources_on_customer
+    {
+        public StripeList<StripeSource> SourceListAll { get; }
+        public StripeList<StripeSource> SourceListCard { get; }
+        public StripeList<StripeSource> SourceListBitcoin { get; }
+
+        public listing_sources_on_customer()
+        {
+            var sourceService = new StripeSourceService(Cache.ApiKey);
+            var customerService = new StripeCustomerService(Cache.ApiKey);
+
+            // Create customer
+            var CustomerCreateOptions = new StripeCustomerCreateOptions
+            {
+                Email = "source-list@example.com",
+            };
+            var Customer = customerService.Create(CustomerCreateOptions);
+
+            // Create card source and attach it to customer
+            var SourceCardCreateOptions = new StripeSourceCreateOptions
+            {
+                Type = StripeSourceType.Card,
+                Token = "tok_visa"
+            };
+            var SourceCard = sourceService.Create(SourceCardCreateOptions);
+
+            var SourceAttachOptions = new StripeSourceAttachOptions
+            {
+                Source = SourceCard.Id
+            };
+            SourceCard = sourceService.Attach(Customer.Id, SourceAttachOptions);
+
+            // Create bitcoin source and attach it to customer
+            var SourceBitcoinCreateOptions = new StripeSourceCreateOptions
+            {
+                Type = StripeSourceType.Bitcoin,
+                Amount = 1000,
+                Currency = "usd",
+                Owner = new StripeSourceOwner
+                {
+                    Email = "jenny.rosen+fill_now@example.com",
+                },
+            };
+            var SourceBitcoin = sourceService.Create(SourceBitcoinCreateOptions);
+
+            SourceAttachOptions.Source = SourceBitcoin.Id;
+            SourceBitcoin = sourceService.Attach(Customer.Id, SourceAttachOptions);
+
+            // List sources on customer
+            SourceListAll = sourceService.List(Customer.Id);
+
+            var SourceListOptions = new StripeSourceListOptions
+            {
+                Type = StripeSourceType.Card
+            };
+            SourceListCard = sourceService.List(Customer.Id, SourceListOptions);
+
+            SourceListOptions.Type = StripeSourceType.Bitcoin;
+            SourceListBitcoin = sourceService.List(Customer.Id, SourceListOptions);
+        }
+
+        [Fact]
+        public void list_all_should_have_both_sources()
+        {
+            SourceListAll.Data.Count.Should().Be(2);
+
+            var SourceCard = SourceListAll.Data[0];
+            SourceCard.Type.Should().Be(StripeSourceType.Card);
+            SourceCard.Card.Brand.Should().Be("Visa");
+
+            var SourceBitcoin = SourceListAll.Data[1];
+            SourceBitcoin.Type.Should().Be(StripeSourceType.Bitcoin);
```

Comment:
Shouldn't bitcoin be first? Aren't sources returned with most recent first? Or is it due to the default source always being first even if added before and this being undocumented?

Reply:
Yes, the card source is attached first and becomes the default source.

It's definitely not great practice to rely on that behavior in a test, and I should probably not have hardcoded the indices, but it's probably not worth putting in too much effort in those tests since we plan to 🔥  them and use stripe-mock Soon™️.